controllers/update.go

```diff
@@ -496,6 +497,10 @@ func (r *ComponentDetectionQueryReconciler) updateComponentStub(req ctrl.Request
 
 func getComponentName(gitSource *appstudiov1alpha1.GitSource) string {
 	repoUrl := gitSource.URL
+	// If the repository URL ends in a forward slash, remove it to avoid issues with parsing the repository name
+	if string(repoUrl[len(repoUrl)-1]) == "/" {
```

Comment:
Do we need to check if len(repoURL) > 0 ?

Reply:
`gitSource.URL` is a required field for Component CRs, so it shouldn't ever be empty

Reply:
Updated